common/server.cpp

```diff
@@ -168,6 +170,15 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
     event.mutable_user_info()->CopyFrom(session->copyUserInfo(true, true, true));
     locker.unlock();
     
+    // check if client id exists (older client compatibility)
+    if (clientid.isEmpty()){
+        // client id is empty, either out dated client or client has been modified
+    }
+    else {
+        // update users database table with client id
+        databaseInterface->updateUsersClientID(name, clientid);
```

Comment:
I'm saying that whatever gets updated post-login should happen in one call
with 0 or 1 lock instead of X calls with a lock each

On Fri, Aug 7, 2015 at 8:49 AM woogerboy21 <notifications@github.com> wrote:

> In common/server.cpp
> <https://github.com/Cockatrice/Cockatrice/pull/1340#discussion_r36515010>:
>
> > @@ -168,6 +170,15 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
> >      event.mutable_user_info()->CopyFrom(session->
>
> copyUserInfo(true, true, true));
> >      locker.unlock();
> >
> > +    // check if client id exists (older client compatibility)
> > +    if (clientid.isEmpty()){
> > +        // client id is empty, either out dated client or client has been modified
> > +    }
> > +    else {
> > +        // update users database table with client id
> > +        databaseInterface->updateUsersClientID(name, clientid);
>
> Are you saying you want to combine all the update calls to the DB into a
> single function / db call? Cause simply calling the DB post logic with
> multiple function calls doesnt change things a whole lot in terms of
> performance and questionably would increase the amount of time the user
> table gets locked when the updates are occuring.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/1340/files#r36515010>.
>